contrib/init.d/pbs_mom.in

```diff
@@ -28,12 +29,30 @@ else
 fi
 
 pidof_pbs_mom() {
-	pid=`pidof pbs_mom`
-	if [ $? -eq 0 ]; then
-		echo $pid
-	else
-		echo "-1"
-	fi
+    pids=`pidof pbs_mom`
+    if [ $? -ne 0 ]; then
+        echo "-1"
```

Comment:
i'm aware. my point was that you could use the exitcode instead of magic `-1` value. i was not suggesting to return the pid as the exitcode; only that your checks like `if [ $pid -ne -1 ]; then` could be `if [ $? -eq 0 ]; then`.

(and the real problem being that bash doesn't support functions, only commands)